src/common/metadata-utils/index.ts

```diff
@@ -503,7 +504,153 @@ Issue tracking: https://github.com/forcedotcom/cli/issues/2426`)
+          const nameA = a.fullName ? a.fullName[0] : a.name ? a.name[0] : '';
+          const nameB = b.fullName ? b.fullName[0] : b.name ? b.name[0] : '';
+          return nameA.localeCompare(nameB);
+        });
+
+        labels.map(label => {
+          const name = label.fullName ? label.fullName[0] : label.name ? label.name[0] : '';
+          const value = label.value ? label.value[0] : '';
+          const shortDesc = value.length > 40 ? value.substring(0, 40) + '...' : value;
+
+          choices.push({
+            value: name,
+            title: name,
+            description: shortDesc
+          });
+        });
+      }
+
+      const labelSelectRes = await prompts({
+        type: 'multiselect',
+        message: 'Please select the Custom Labels you want to extract from translations',
+        choices: choices
+      });
+
+      return labelSelectRes.value;
+
+    } catch (err: any) {
+      console.error('Error while processing custom labels:', err.message);
+      throw err;
+    }
+  }
+
+  public static async promptLwcComponent() {
+    try {
+      const lwcMetaFiles = await glob('**/lwc/*/*.js-meta.xml');
+
+      if (lwcMetaFiles.length === 0) {
+        throw new Error('No Lightning Web Components found in the project');
+      }
+
+      const componentsInfo: Array<any> = [];
+      for (const metaFile of lwcMetaFiles) {
+        try {
+          const xmlContent = await fs.readFile(metaFile, 'utf8');
+          const parsedXml = await parseStringPromise(xmlContent);
+
+          const pathParts = metaFile.split('/');
+          const componentName = pathParts[pathParts.length - 1].replace('.js-meta.xml', '');
+
+          let masterLabel = componentName;
+
+          if (parsedXml.LightningComponentBundle &&
+            parsedXml.LightningComponentBundle.masterLabel &&
+            parsedXml.LightningComponentBundle.masterLabel.length > 0) {
+            masterLabel = parsedXml.LightningComponentBundle.masterLabel[0];
+          }
+
+          componentsInfo.push({
+            name: componentName,
+            label: masterLabel,
+            path: metaFile
+          });
+        } catch (err: any) {
+          console.warn(`Could not parse meta file: ${metaFile}`, err.message);
+        }
+      }
+
+      componentsInfo.sort((a, b) => a.name.localeCompare(b.name));
+
+      const choices = componentsInfo.map(component => ({
+        value: component.name,
+        title: component.label,
+        description: `Name: ${component.name}`
+      }));
+
+      const componentSelectRes = await prompts({
+        type: 'select',
+        name: 'value',
+        message: 'Select a Lightning Web Component to extract custom labels from',
+        choices: choices
+      });
 
+      return componentSelectRes.value;
+    } catch (err: any) {
+      console.error('Error while finding LWC components:', err.message);
+      throw err;
+    }
+  }
+
+  public static async promptExtractionMethod() {
```

Comment:
this method will be used only in your command, please move it into your command :)